.github/workflows/haskell-ci.yml

```diff
@@ -118,9 +118,11 @@ jobs:
           repository hackage.haskell.org
             url: http://hackage.haskell.org/
           EOF
+          cat $CABAL_CONFIG
           if $HEADHACKAGE; then
-          echo "allow-newer: $($HCPKG list --simple-output | sed -E 's/([a-zA-Z-]+)-[0-9.]+/*:\1/g')" >> $CABAL_CONFIG
-          cat >> $CABAL_CONFIG <<EOF
+          touch cabal.project
+          echo "allow-newer: $($HCPKG list --simple-output | sed -E 's/([a-zA-Z-]+)-[0-9.]+/*:\1,/g')" >> cabal.project
```

Comment:
we don't need this here. we don't need `allow-newer` to run `cabal v2-update` nor for `sdist`.

Add generation of `allow-newer` to the second `cabal.project` which is used with tests.

Reply:
To clarify, `allow-newer` were written to `cabal.config` because it was convenient. `repostory` stanza should stay in `cabal.config`.

Reply:
Good suggestion. I've pushed a new version which implements this idea, which has the fortunate side effect of simplifying the implementation in the Travis backend.